consensus/propagation/propagator.go

```diff
@@ -13,11 +14,16 @@ type Propagator interface {
 	ProposeBlock(proposal *types.Proposal, parts *types.PartSet, txs []proptypes.TxMetaData)
 	AddCommitment(height int64, round int32, psh *types.PartSetHeader)
 	Prune(committedHeight int64)
-	SetConsensusRound(height int64, round int32)
+	SetHeightAndRound(height int64, round int32)
 	StartProcessing()
 	SetProposer(proposer crypto.PubKey)
 	GetPartChan() <-chan types.PartInfo
-	GetProposalChan() <-chan types.Proposal
+	GetProposalChan() <-chan ProposalAndFrom
+}
+
+type ProposalAndFrom struct {
```

Comment:
so that we send the proposal along with the peer who sent it. This is mainly used for the consensus peer state to be updated with latest proposals